harmonica/meshes.py

```diff
@@ -0,0 +1,212 @@
+"""
+Meshes and layers of simple geometric elements.
+"""
+import numpy as np
+import xarray as xr
+from verde.coordinates import spacing_to_shape, check_region
+
+from . import geodetic_to_spherical
+
+
+def tesseroids_layer(
+    region,
+    spacing=None,
+    shape=None,
+    adjust="spacing",
+    top=None,
+    bottom=None,
+    density=None,
+    coordinates="spherical",
+    region_centers=True,
+):
+    """
+    Create a layer of tesseroids.
+
+    The layer can be specified by either the number of tesseroids in each dimension (the
+    *shape*) or by the size of each tesseroid (the *spacing*).
+
+    The layer will be a :class:`xarray.Dataset` with ``longitude`` and ``latitude``
+    coordinates and `top`, `bottom` and `density` as data.
+    The ``longitude`` and ``latitude`` will be the coordinates of the center of each
+    tesseroid on the layer in geocentric spherical coordinates.
+    The ``top`` and ``bottom`` parameters can be defined either in a geocentric
+    spherical or in a geodetic coordinate system.
+
+    .. warning :
+        If ``top`` and ``bottom`` parameters are defined under a geodetic coordinate
+        system, they will be converted to a spherical geocentric one.
+        Because tesseroids cannot be defined on a geodetic coordinate system, the
+        latitude coordinates will remain unchanged.
+
+    If the given region is not divisible by the desired spacing, either the region or
+    the spacing will have to be adjusted. By default, the spacing will be rounded to the
+    nearest multiple. Optionally, the East and North boundaries of the region can be
+    adjusted to fit the exact spacing given. See the examples below.
+
+    Parameters
+    ----------
+    region : list = [W, E, S, N]
+        The boundaries of a given region in geocentric spherical coordinates in degrees.
+    spacing : float, tuple = (d_lat, d_lon), or None
```

Comment:
`cell_size` is probably a better term than `spacing` for meshes

Reply:
I kept `spacing` because the `adjust` argument of `verde.coordinates.spacing_to_shape` must be either `shape` or `spacing`.
I don't quite like it for the tesseroids, so we could change the argument name and internally pass `spacing` to `spacing_to_shape` when `adjusts="spacing"`.

PS: I want the legal values for `adjust` and the `cell_size`/`spacing` argument are in agreement.